.github/workflows/optic.yml

```diff
@@ -0,0 +1,28 @@
+name: optic
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  diff-all:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: opticdev/action@v1
+        with:
+
+          # A GitHub token with access to create comments on pull requests
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
This is tricky because PRs from forks won't have access to secrets, so this setup won't work for those cases. This is something we've hit with a lot of our feedback workflows, and the workaround is to run the workflow with the `pull_request_target` trigger instead, which runs from the base branch with access to secrets. See [this](https://github.com/backstage/backstage/blob/5a432881d7d81ab12a91e2927b1d457a028597bd/.github/workflows/automate_changeset_feedback.yml#L3) workflow for example.

When using `pull_request_target` it is extremely important to not bring in and execute any user code. Since we want to check out the code in the PR branch it puts the burden on `opticdev/action` to not allow for any injection of code, or in any other way leak secrets. For example, as far as I know [this](https://github.com/opticdev/action/blob/d1a9e5a7bc54d74acaf5afff8bd8829c60f1fbc0/src/action.ts#L158) would be fairly trivial to inject your own code into by providing a `.npmrc` with some custom registry configuration.

This does leave us with a bit of an issue, where Optic might not be intended for use with a `pull_request_target` trigger and we'd need to consider workarounds.

Reply:
Thanks for this, I hadn't realized how much more complicated this would be. How would I go about verifying if Optic supports/wont leak secrets with the `pull_request_target`? Are any of our other actions node/JS actions?

Reply:
Unless the Optic action is specifically built to allow for `pull_request_target` I wouldn't go down the route of trying to make it safe, and probably not even then tbh 😅 . It's a large surface area and even if the current state is alright we don't know that vulnerabilities won't be introduced in the future.

I think for this case we'll want to use [this](https://github.com/backstage/backstage/blob/5a432881d7d81ab12a91e2927b1d457a028597bd/.github/workflows/pr-review-comment.yaml#L4-L8) pattern where we upload the result of the scan to an artifact of a read-only workflow. The completion of that workflow then triggers a separate workflow where we post the results. This reduces the surface area of the code a lot and crucially eliminates almost all user input, we just need to make sure nothing that can be put in the artifact could lead to any form of injection in the triggered workflow.